src/plugins/mdraid.c

```diff
@@ -193,8 +194,16 @@ static GHashTable* parse_mdadm_vars (const gchar *str, const gchar *item_sep, co
         if (g_strv_length ((gchar **) key_val) == 2) {
             /* we only want to process valid lines (with the separator) */
             /* only use the first value for the given key */
-            if (!g_hash_table_contains (table, g_strstrip (key_val[0])))
-                g_hash_table_insert (table, g_strstrip (key_val[0]), g_strstrip (key_val[1]));
+            if (!g_hash_table_contains (table, g_strstrip (key_val[0]))) {
+                vals = g_strsplit (key_val[1], "<--", 2);
+                if (g_strv_length ((gchar **) vals) == 2)
```

Comment:
ok! I literally have no idea what I'm doing in C so I took the safe path by mimicking the existing code as far as I could, but I'll figure that out tomorrow =)